core/src/model.jl

```diff
@@ -118,6 +118,8 @@ function Model(config::Config)::Model
 
     # Synchronize level with storage
     set_current_basin_properties!(du0, u0, parameters, t0)
+    parameters.basin.level_prev .=
```

Comment:
Would be good to inline comment (or in the PR description) as to why we now need to keep track of `level_prev`.